partiql-eval/src/plan.rs

```diff
@@ -187,6 +187,40 @@ impl EvaluatorPlanner {
                 let to = self.plan_values(*expr.to);
                 Box::new(EvalBetweenExpr { value, from, to })
             }
+            ValueExpr::SimpleCase(e) => {
+                let cases = e
+                    .cases
+                    .into_iter()
+                    .map(|case| {
+                        (
+                            self.plan_values(ValueExpr::BinaryExpr(
+                                BinaryOp::Eq,
+                                e.expr.clone(),
+                                case.0,
+                            )),
+                            self.plan_values(*case.1),
+                        )
+                    })
+                    .collect();
+                let default = e
+                    .default
+                    .as_ref()
+                    .map(|default| self.plan_values(*default.clone()));
+                // Here, rewrite `SimpleCaseExpr`s as `SearchedCaseExpr`s
+                Box::new(EvalSearchedCaseExpr { cases, default })
+            }
+            ValueExpr::SearchedCase(e) => {
+                let cases = e
+                    .cases
+                    .into_iter()
+                    .map(|case| (self.plan_values(*case.0), self.plan_values(*case.1)))
+                    .collect();
+                let default = e
+                    .default
+                    .as_ref()
+                    .map(|default| self.plan_values(*default.clone()));
+                Box::new(EvalSearchedCaseExpr { cases, default })
```

Comment:
Could also make the `default` a non-option type by following what's mentioned in the SQL-92 spec (section 6.9 <case expression> pg 141):

```
 6) If an <else clause> is not specified, then ELSE NULL is implicit.
```

Reply:
Updated in 7de9988dd95e286ba8e7d92198b675f7d2ea4168 to use implicit `ELSE NULL` when no `ELSE` clause is provided.